File: day1.hs
import System.IO  
import Control.Monad

main = do  
        let list = []
        handle <- openFile "input/day1.txt" ReadMode
        contents <- hGetContents handle
        let singlewords = words contents
            list = f singlewords
            -- First part
            first_part = filter p (zip list (tail list))
            -- Second part
            second_part = filter p (zip list (drop 3 list))
        print (length first_part)
        print (length second_part)
        hClose handle

f :: [String] -> [Int]
f = map read

p ::  Ord a => (a, a) -> Bool
p (a, b) = a < b

File: day1.js
const fs = require('fs');

function zip(arrays) {
    return arrays[0].map(function(_,i){
        return arrays.map(function(array){return array[i]})
    });
}

// part 1
in1 = fs.readFileSync('input/day1.txt').toString().split('\r\n').map(function(item) {
    return parseInt(item);
});
in2 = fs.readFileSync('input/day1.txt').toString().split('\r\n').map(function(item) {
    return parseInt(item);
});
in2.splice(0, 1);
sol1 = zip([in1, in2]).filter(a => {
	return a[0] < a[1];
});

// second part
in2.splice(0, 2);
sol2 = zip([in1, in2]).filter(a => {
	return a[0] < a[1];
})

console.log(sol1.length, sol2.length);